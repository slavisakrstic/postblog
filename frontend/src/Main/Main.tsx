import { lazy, Suspense } from "react";
import { Switch, Route, Redirect } from "react-router";
import { Router } from "react-router-dom";

// Routes
import { POSTS, POST_DETAILS } from "../constants/routes";

import history from "../core/history"

// Custom components
import Layout from "../Layout";
import Loading from "../Loading";

const ViewPosts = lazy(() => import("../Post"));
const ViewPost = lazy(() => import("../PostDetails"));

const Main = () => {
  return (
    <Router history={history}>
      <Switch>
        <Suspense fallback={<Loading />}>
          <Layout>
            <Route exact path="/" render={() => <Redirect to={POSTS} />} />
            <Route exact path={POSTS} component={ViewPosts} />
            <Route exact path={POST_DETAILS} component={ViewPost} />
          </Layout>
        </Suspense>
      </Switch>
    </Router>
  );
};

export default Main;