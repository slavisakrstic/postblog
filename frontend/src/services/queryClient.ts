import { QueryClient } from "react-query";

const defaultConfiguration = {
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnWindowFocus: false
    }
  }
};

export const createQueryClient = (config = defaultConfiguration) =>
  new QueryClient(config);
const defaultQueryClient = createQueryClient();

export { defaultQueryClient as queryClient };