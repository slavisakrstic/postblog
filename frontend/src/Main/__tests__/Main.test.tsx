import { render } from "@testing-library/react";
import { createQueryClient } from "../../services/queryClient";
import { QueryClientProvider } from 'react-query';
import axios from "axios";
import Main from "../Main";

jest.mock("axios");

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => { }),
      },
    };
  },
}));

describe("Main", () => {
  const queryClient = createQueryClient();

  it("should render main component", async () => {
    const getMock = jest.fn().mockResolvedValue({
      data: [{
        id: 1,
        title: "title 1",
        blog: "blog 1",
        numberOfComments: 2
      }]
    });
    axios.get = getMock;

    expect(render(<QueryClientProvider client={queryClient}><Main /></QueryClientProvider>)).toMatchSnapshot();
  });
})