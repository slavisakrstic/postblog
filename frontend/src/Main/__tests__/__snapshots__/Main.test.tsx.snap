// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render main component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
        style="display: none;"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-5 mui-fixed MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
              data-testid="general-title"
            >
              __general.title
            </h6>
          </div>
        </header>
        <div
          class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-2"
        >
          <div
            class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-3 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
            />
            <div
              class="makeStyles-drawerContainer-6"
            >
              <ul
                class="MuiList-root MuiList-padding"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  color="primary"
                  data-testid="new-post"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root makeStyles-textColor-7"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      __post.create
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </ul>
            </div>
          </div>
        </div>
        <main
          class="makeStyles-content-4"
        />
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <div
          class="MuiBox-root MuiBox-root-9"
        >
          <h1
            class="MuiTypography-root makeStyles-loader-8 MuiTypography-h4 MuiTypography-gutterBottom"
          >
            <div
              class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
              data-testid="linear-progress-id"
              role="progressbar"
            >
              <div
                class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
              />
              <div
                class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
              />
            </div>
          </h1>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
      style="display: none;"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-5 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
            data-testid="general-title"
          >
            __general.title
          </h6>
        </div>
      </header>
      <div
        class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-2"
      >
        <div
          class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-3 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          />
          <div
            class="makeStyles-drawerContainer-6"
          >
            <ul
              class="MuiList-root MuiList-padding"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                color="primary"
                data-testid="new-post"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root makeStyles-textColor-7"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    __post.create
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </ul>
          </div>
        </div>
      </div>
      <main
        class="makeStyles-content-4"
      />
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <div
        class="MuiBox-root MuiBox-root-9"
      >
        <h1
          class="MuiTypography-root makeStyles-loader-8 MuiTypography-h4 MuiTypography-gutterBottom"
        >
          <div
            class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
            data-testid="linear-progress-id"
            role="progressbar"
          >
            <div
              class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
            />
            <div
              class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
            />
          </div>
        </h1>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
